src/fighter_state.rs

```diff
@@ -1433,23 +1443,19 @@ fn holding(
 
 fn melee_attacking(
     mut commands: Commands,
-    mut fighters: Query<(
-        Entity,
-        Option<&mut MeleeAttacking>,
-        Option<&Player>,
-        Option<&Enemy>,
-        &AvailableAttacks,
-        &mut LinearVelocity,
-        &Facing,
-    )>,
-    mut melee_weapons: Query<(
-        Entity,
-        &Parent,
-        &mut Animation,
-        &MeleeWeapon,
-        &mut Transform,
-        &mut TextureAtlasSprite,
-    )>,
+    mut fighters: Query<
+        (
+            Entity,
+            Option<&mut MeleeAttacking>,
+            Option<&Player>,
+            Option<&Enemy>,
+            &AvailableAttacks,
+            &mut LinearVelocity,
+            &Facing,
+        ),
+        Without<MeleeWeapon>,
```

Comment:
Is this `Without` filter necessary?

Same for the `Without<ProjectileWeapon>` below.

Reply:
Just checked and they're not, they were before, but with the Attached, not anymore.